### PR TITLE
NodeBuilder: Missing property in requires type definition

### PR DESCRIPTION
**Pull request by martinRenou**
Used here for example: https://github.com/mrdoob/three.js/blob/dev/examples/jsm/nodes/materials/nodes/StandardNode.js#L229 and https://github.com/mrdoob/three.js/blob/dev/examples/jsm/nodes/materials/nodes/StandardNode.js#L241

cc. @sunag

### CODE REVIEW

File: examples/jsm/nodes/core/NodeBuilder.d.ts [hunk 1]
```diff
@@ -24,6 +24,8 @@ export class NodeBuilder {
 		color: boolean[];
 		lights: boolean;
 		fog: boolean;
+		transparent: boolean;
+		irradiance: boolean;
```
Comment (sunag) [inline examples/jsm/nodes/core/NodeBuilder.d.ts:28]: Irradiance in NodeBuilder?

**Comment by sunag** [comment]
Ahh ok. You can add these properties here: https://github.com/mrdoob/three.js/blob/fb6cf2dc86518cef9a570bfa6be40a46edbfe016/examples/jsm/nodes/core/NodeBuilder.js#L63

**Comment by martinRenou** [comment]
> Ahh ok. You can add these properties here:

Will do! 

Should their default value be false?

File: examples/jsm/nodes/core/NodeBuilder.d.ts (hunk 1 above)
Comment (martinRenou) [reply examples/jsm/nodes/core/NodeBuilder.d.ts:28]: Yes, it seems to be used here: https://github.com/mrdoob/three.js/blob/dev/examples/jsm/nodes/materials/nodes/StandardNode.js#L229

But there is a big chance I misunderstand something. I don't quite understand this code base and the logic yet.

**Comment by sunag** [comment]
> Should their default value be false?

Yes, yes. This values is really defined in `.build()` process.

**Comment by martinRenou** [comment]
Ok :)

@sunag now that I have a bit of you attention, say I want to discuss with you some of the code base and improvements I would like to push on actual code (not only TypeScript this time). Is there a discourse discussion where we could chat or something equivalent?

**Comment by sunag** [comment]
> @sunag now that I have a bit of you attention, say I want to discuss with you some features or improvements I would like to push on actual code (not only TypeScript this time). Is there a discourse discussion where we could chat or something equivalent?

We can use private message of the twitter for chat? You can add me @sea3dformat

**Comment by martinRenou** [comment]
> We can use private message of the twitter for chat? You can add me @sea3dformat

Ahah, great, I followed you a week ago I think. I did not realize @sea3dformat was actually sunag who implemented the amazing NodeMaterials, I would have followed you sooner if I knew :D

I am @martinRenou on Twitter, I guess you only can send the first private message (I don't see any button for sending a PM).

**Comment by mrdoob** [comment]
@sunag looks good now?

File: examples/jsm/nodes/core/NodeBuilder.js [hunk 2]
```diff
@@ -64,7 +64,9 @@ function NodeBuilder() {
 		uv: [],
 		color: [],
 		lights: false,
-		fog: false
+		fog: false,
+		transparent: false,
+		irradiance: false,
```
Comment (sunag) [inline examples/jsm/nodes/core/NodeBuilder.js:69]: The last comma in `irradiance` maybe show warns in the compiler.

**Review by sunag — APPROVED** [review]
@martinRenou Thanks again. Maybe I remove `irradiance` in the future, I think that this property should have relation only with PBR materials...

**Comment by sunag** [comment]
> @sunag looks good now?

Yes, yes :+1: :+1:

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**